server/src/main/java/io/spine/server/aggregate/Aggregate.java

```diff
@@ -392,15 +420,33 @@ protected final String missingTxMessage() {
     /**
      * Transforms the current state of the aggregate into the {@link Snapshot} instance.
      *
+     * <p>If the {@linkplain #isTransactionInProgress() transaction is in progress}, the state,
+     * version and lifecycle are taken from the transactional data.
+     *
      * @return new snapshot
      */
     final Snapshot toSnapshot() {
-        Any state = AnyPacker.pack(state());
+        S state;
+        Version version;
+        LifecycleFlags lifecycle;
+        if (isTransactionInProgress()) {
+            AggregateTransaction tx = (AggregateTransaction) tx();
```

Comment:
Please use generic parameters having: 

```java
            AggregateTransaction<?, ?, ?> tx = (AggregateTransaction<?, ?, ?>) tx();
```
It would eliminate the IDEA's warning on generics.